nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -198,6 +198,200 @@ public void downcastInstantiation() {
         .doTest();
   }
 
+  // assignment tree tests
+  @Test
+  public void genericsChecksForAssignments() {
```

Comment:
Generally, we need to clean up these tests.  It should be relatively obvious to a reader what the purpose of each line is.  So we should split up the tests into more methods as needed, give them meaningful names, and remove redundant tests